### DISCUSSION TITLE
wix0103 error in Bootstrapper project - WiX4

### DISCUSSION THREAD

**Discussion by pas059** [wixtoolset/issues#7619] [Questions] [answered]
Hello,
I currently try to upgrade Wix3 projects to Wix 4. To do this i use HeatWave  FireGiant's Visual Studio extension. For the installer projects this seems to work. But for a Bootstrapper project that embed .net472 installer, i encounter the following error:

error WIX0103: Cannot find the WixBundlePayload file 'SourceDir\C:\Logiciels\Microsoft\dotNet 4.x\NDP472-KB4054530-x86-x64-AllOS-ENU.exe'. The following paths were checked: SourceDir\C:\Logiciels\Microsoft\dotNet 4.x\NDP472-KB4054530-x86-x64-AllOS-ENU.exe

... but the file exists

Below, the Fragment that is at the origin of the error:
`	<Fragment>
		<util:RegistrySearchRef Id="NETFRAMEWORK45" />

		<WixVariable Id="WixMbaPrereqPackageId" Value="NetFx472Redist" />
		<WixVariable Id="WixMbaPrereqLicenseUrl" Value="$(var.NetFx472EulaLink)" Overridable="yes" />
		<WixVariable Id="NetFx472RedistDetectCondition" Value="NETFRAMEWORK45 &gt;= $(var.NetFx472MinRelease)" Overridable="yes" />
		<WixVariable Id="NetFx472RedistInstallCondition" Value="" Overridable="yes" />
		<WixVariable Id="NetFx472RedistPackageDirectory" Value="C:\Logiciels\Microsoft\dotNet 4.x\" Overridable="yes" />

		<PackageGroup Id="NetFx472RedistEmbedded">
			<ExePackage 
				PerMachine="yes" 
				DetectCondition="!(wix.NetFx472RedistDetectCondition)" 
				InstallCondition="!(wix.NetFx472RedistInstallCondition)" 
				Id="NetFx472Redist" Vital="yes" Permanent="yes" 
				Protocol="netfx4" 
				LogPathVariable="NetFx472FullLog" 
				Compressed="yes" 
				Name="!(wix.NetFx472RedistPackageDirectory)NDP472-KB4054530-x86-x64-AllOS-ENU.exe"
				InstallArguments="/q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;" 
				RepairArguments="/q /norestart /repair /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;" 
				UninstallArguments="/uninstall /q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;">
			</ExePackage>
		</PackageGroup>
	</Fragment>
`
This code was slightly updated by HeatWave but worked with Wix3; it's like the string "SourceDir\" was added in path.
Someone can help?

Pascal

**Comment by pas059** [comment, marked as answer]
Hi,
I thnik that i solved this issue by replacing the 'Name' attibute by 'SourceFile' attribute; be careful, the 'SourceFile' attribute has to be a relative path. This gives something like this:
`	<Fragment>
		<util:RegistrySearchRef Id="NETFRAMEWORK45" />

		<WixVariable Id="WixMbaPrereqPackageId" Value="NetFx472Redist" />
		<WixVariable Id="WixMbaPrereqLicenseUrl" Value="$(var.NetFx472EulaLink)" Overridable="yes" />
		<WixVariable Id="NetFx472RedistDetectCondition" Value="NETFRAMEWORK45 &gt;= $(var.NetFx472MinRelease)" Overridable="yes" />
		<WixVariable Id="NetFx472RedistInstallCondition" Value="" Overridable="yes" />
		<WixVariable Id="NetFx472RedistPackageDirectory" Value="..\Install_Common\" Overridable="yes" />
		
		<PackageGroup Id="NetFx472RedistEmbedded">
			<ExePackage 
				PerMachine="yes" 
				DetectCondition="!(wix.NetFx472RedistDetectCondition)" 
				InstallCondition="!(wix.NetFx472RedistInstallCondition)" 
				Id="NetFx472RedistEmbedded" Vital="yes" Permanent="yes" 
				Protocol="netfx4" 
				LogPathVariable="NetFx472FullLog" 
				Compressed="yes" 
				SourceFile ="!(wix.NetFx472RedistPackageDirectory)NDP472-KB4054530-x86-x64-AllOS-ENU.exe"
				InstallArguments="/q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;" 
				RepairArguments="/q /norestart /repair /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;" 
				UninstallArguments="/uninstall /q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;">			
			</ExePackage>
		</PackageGroup>

	</Fragment>
`
now the bootstrapper is generated; there are some remarks compared to WiX 3 but this is another topics.

**Reply by robmen** [reply]
I am surprised using the `Name` attribute in v3 worked like that, but the code was a bit goofy. Probably a case of us cleaning up the code and catching an error case that slipped through in v3.

**Reply by pas059** [reply]
In fact, i found this code somewhere 2 or 3 years ago, but i didn't remember where
